libass/ass_utils.c

```diff
@@ -116,10 +117,20 @@ void *ass_try_realloc_array(void *ptr, size_t nmemb, size_t size)
 
 void ass_msg(ASS_Library *priv, int lvl, const char *fmt, ...)
 {
+#if ENABLE_THREADS
+    if (!priv->thread_safe_cb)
+        pthread_mutex_lock(&priv->log_mutex);
+#endif
```

Comment:
Potential cosmetic issue: our default log handler uses several printf calls and is listed as thread safe. Individual calls from different messages might end up interleaved.

Strictly speaking, albeit only a theoretical concern for any remotely relevant platform as far as i’m aware: ISO C99 to my knowledge doesn't guarantee thread-safety of `(f))printf`. [POSIX explicitly guarantees this](https://pubs.opengroup.org/onlinepubs/9699919799/functions/flockfile.html) and provides `_unlocked` non-thread-safe variants for when custom locking is used and print perf relevant.

Reply:
Yeah, this interleaving does happen in practice, but I'm not particularly concerned about it (since as you say it's cosmetic). I've found the overhead of locking to actually be meaningful in some cases (esp. when people put nonstandard stuff in the `Effect` field, since that logs on every frame).

Reply:
yeah, i’m not too concerned either; mostly just brought it up again since the old thread ended without a conclusion on `flockfile`.  
If interleaving turns out too annoying in practice we can still change this later